src/dataServices/dynamoDbParamBuilder.ts

```diff
@@ -54,20 +57,29 @@ export default class DynamoDbParamBuilder {
                 ':pendingDeleteStatus': DOCUMENT_STATUS.PENDING_DELETE,
                 ':currentTs': currentTs,
                 ':futureEndTs': futureEndTs,
+                ':resourceType': resourceType,
             });
         }
 
         return params;
     }
 
-    static buildGetResourcesQueryParam(id: string, maxNumberOfVersions: number, projectionExpression?: string) {
+    static buildGetResourcesQueryParam(
+        id: string,
+        resourceType: string,
+        maxNumberOfVersions: number,
+        projectionExpression?: string,
+    ) {
         const params: any = {
             TableName: RESOURCE_TABLE,
             ScanIndexForward: false,
             Limit: maxNumberOfVersions,
+            FilterExpression: '#r = :resourceType',
             KeyConditionExpression: 'id = :hkey',
+            ExpressionAttributeNames: { '#r': 'resourceType' },

```

Comment:
do you want to use quotes here?
```suggestion
            ExpressionAttributeNames: { '#r': resourceType },

```

Reply:
Yep, quote is needed here. In this case we want to map `#r` to the string `resourceType`

Similar to here in relation to `year`
```
var params = {
    TableName : "Movies",
    KeyConditionExpression: "#yr = :yyyy",
    ExpressionAttributeNames:{
        "#yr": "year"
    },
    ExpressionAttributeValues: {
        ":yyyy": 1985
    }
};
```
https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/GettingStarted.NodeJs.04.html

Reply:
Ahh thanks I didn't see the `name` and `values` sections